sad_comparison_functions.py

```diff
@@ -72,6 +90,50 @@ def get_pred_multi_dists(S, dist_name, pars):
         print "Error: distribution not recognized."
         pred = None
     return pred
+
+def get_loglik_multi_dists(ab, dist_name, pars):
+    """Returns the log-likelihood given abundances, 
+    
+    the designated distribution, and the parameters.
+    
+    """
+    dist = 0
+    if dist_name == 'logser': dist = sd.logser
+    elif dist_name == 'geom': dist = sd.geom
+    elif dist_name == 'zipf': dist = sd.zipf
+    if dist: # If one of the above three cases:
+        loglik = sum(dist.logpmf(ab, pars))
+    else:
+        if dist_name == 'negbin':
+            n, p = pars
+            loglik = sum(np.log(md.nbinom_lower_trunc.pmf(ab, n, p)))
+        elif dist_name == 'pln':
+            mu, sigma = pars
+            loglik = sum(np.log(md.pln.pmf(ab, mu, sigma, True)))
+    return loglik
+
+def get_ks_multi_dists(ab, dist_name, pars):
+    """Returns the K-S statistic given abundances, 
+    
+    the designated distribution, and the parameters.
+    
+    """
+    ab = sorted(ab)
+    emp_cdf = (np.arange(1, len(ab) + 1) - 0.5) / len(ab)  
+    dist = 0
+    if dist_name == 'logser': dist = sd.logser
+    elif dist_name == 'geom': dist = sd.geom
+    elif dist_name == 'zipf': dist = sd.zipf
+    if dist: # If one of the above three cases:
+        ks = max(abs(emp_cdf - np.array([dist.cdf(x, pars) for x in ab])))
+    else:
+        if dist_name == 'negbin':
+            n, p = pars
+            ks = max(abs(emp_cdf - np.array([md.nbinom_lower_trunc.cdf(x, n, p) for x in ab])))
+        elif dist_name == 'pln':
+            mu, sigma = pars
+            ks = max(abs(emp_cdf - np.array([md.pln.cdf(x, mu, sigma, True) for x in ab])))
+    return ks
```

Comment:
Similar uses of dictionaries to the one described above are also possible for both of these functions.